### ISSUE TITLE
Help us translate the app

### ISSUE THREAD

**Issue by CDRussell** [duckduckgo/Android#559]
## New languages
We will be adding additional languages to the app and we're needing help in improving our translations. 

For now, we're only looking for improvements to the list of languages identified and are not looking to add completely new languages yet. We may request help on doing this in future, though.

Do you know any of the following languages, and want to help us improve? If so, please see https://github.com/duckduckgo/Android/blob/develop/TRANSLATIONS.md for instructions on how you can help us improve the app.

### List of languages

- Bulgarian (bg)
- Croatian (hr)
- Czech (cs)
- Danish (da)
- Dutch (nl)
- Finnish (fi)
- French (fr)
- German (de)
- Greek (el)
- Hungarian (hu)
- Italian (it)
- Lithuanian (lt)
- Norwegian bokmål (nb)
- Polish (pl)
- Portuguese (pt)
- Romanian (ro)
- Russian (ru)
- Slovak (sk)
- Slovenian (sl)
- Spanish (es)
- Swedish (sv)

**Comment by coffeeplanter** [comment]
Hi, I've seen the current Russian translation. It's pretty accurate and tidy, in my opinion. But I'll make a pull-request.
And... why to translate string resources marked as non translatable? )

**Comment by CDRussell** [comment]
non translatable should indicate things like the company name, where DuckDuckGo doesn't need to be translated. however, it's possible some things have been incorrectly marked as non-translatable that should be translated, or things are currently translated that shouldn't be.

A tidy-up PR would be awesome! 🙏

**Comment by renanferrari** [comment]
> not looking to add completely new languages yet

Would it be ok to add language variants, since it may not be considered a "completely new language"?

I'm asking this because I believe **Brazilian Portuguese** would be a great addition in the list, based on the fact that it is – by far – the most widely spoken Portuguese variant in the world, while **European Portuguese** is only the fourth:

Rank | Country | Number of Portuguese Speakers (Est.)
-- | -- | --
**1** | **Brazil** | **204,000,000**
2 | Angola | 20,455,100
3 | Mozambique | 14,400,000
**4** | **Portugal** | **10,320,000**
5 | Guinea-Bissau | 272,400
6 | Cape Verde | 215,000
7 | Sao Tome and Principe | 199,910
8 | Macao | 6,800

[Source](https://www.worldatlas.com/articles/portuguese-speaking-countries.html)

I believe this is the most relevant data to take into account, so I'm leaving aside more subjective issues like the negative effects on usability or the feeling of unfamiliarity when using the app – besides, I might be biased towards those, since I'm Brazilian myself.

Summing up, if you agree that prioritizing languages spoken by larger audiences is the main factor to be taken into consideration here, then I suggest you approve PR #598, which was opened a few weeks ago. Otherwise, I'd love to know of any other factors that I may not be aware of. Either way, I'll be happy to hear your thoughts on this.

**Comment by geekanamika** [comment]
@CDRussell If you are planning to translate the app to Hindi (Hi) in near future, I would be happy to help. Thanks.

**Comment by subsymbolic** [comment]
We are adding Estonian, Latvian & Turkish support as well as filling in gaps across our already supported languages and would appreciate any help validating these translations too.

See https://github.com/duckduckgo/Android/pull/740 for a list of upcoming changes.

**Comment by cmonfortep** [comment]
Friendly reminder for folks following this issue and future readers. 

We have updated our Translations contribution guidelines. See them here: https://github.com/duckduckgo/Android/blob/develop/TRANSLATIONS.md

Important part is that now we are using an external service that should increase the quality of our translations. So now we only accept contributions to fix actual issues (typos, grammatical errors) and won't accept minor changes that are only focused on tweaking the wording.

**Comment by oldphones68** [comment]
@CDRussell Bro!
Is there any option to contribute if I can't or don't want to use the xml format?
Say for example, I had seen some devs to use something like Google Sheets to translate.
There are the main terms/words/sentences in the left column and people translate on the left columns detivated for each language.

Proobably, for you it may become harder to build from that, but from non-programmers, it may seems more handy.

Hope you'll understand and consider.

Regards!

**Comment by cmonfortep** [comment]
@oldphones68 the only way to contribute to our source code in Github is by opening a Pull request, and that should follow the XML format so we can merge your contribution.

If you are a non-programmer and you spot any major issue with any of our translations, a good idea will be to just open us a bug report via Github Issues. In the description, you can give us some details and propose a new translation. We can take it from there and give feedback to our translation service. Hope that answers your question. Thanks!

**Comment by oldphones68** [comment]
Thank you so much! @cmonfortep

**Comment by Nihal247** [comment]
No Hindi? I think it's required. Hindi is the fourth most spoken language in the World.

**Comment by prashantkumar-cmd** [comment]
I would like to add ' Hindi '  to App.

**Comment by Zoo-M0** [comment]
I'm native Italian speaker, I will look into it for improvements or corrections

**Comment by Hey-Deepak** [comment]
Namaste!

I m native Hindi Speaker and We train student about app development. (
Kotlin + JetCompose )
Want to contribute & want to make this app popular in India.
Firstly, we want to go with Language.
Although, in India everyone use English language for app but still want to
contribute in this issue.

Expecting your reply.

**Comment by BenGeorgeNetto** [comment]
I would like to add Malayalam to the list of supported languages and would like to contribute in case it is added to the app by providing translations. So if you're planning to translate the app to Malayalam would like to help with that

**Comment by CYAXXX** [comment]
I'd like to start translating for Kurdish language, any tips?
